### PR TITLE
First version of ecosystem example with zfit

### PR DESCRIPTION
**Pull request by eduardo-rodrigues**

### CODE REVIEW

File: 06-ecosystem-example.ipynb [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+    "right_tale_data_rare = zfit.Data.from_numpy(obs=obs_bkg, array=comb_bkg_rare_sample.value())\n",
+    "\n",
+    "# set the value of lambda to smth different than we sampled from (for the fit afterwards)\n",
+    "lambda_rare.set_value(-0.003)\n",
+    "\n",
+    "# here we temporarily set the normalization range to the right side only\n",
+    "with comb_bkg_rare.set_norm_range(obs_bkg):\n",
+    "    right_tale_loss = zfit.loss.UnbinnedNLL(comb_bkg_rare, right_tale_data_rare)\n",
+    "    minimizer = zfit.minimize.Minuit(verbosity=7)\n",
+    "    minimizer._use_tfgrad = False\n",
```
Comment (jonas-eschle) [inline 06-ecosystem-example.ipynb]:
```suggestion
    "    minimizer = zfit.minimize.Minuit(verbosity=7, use_minuit_grad=True)\n",
```

File: 06-ecosystem-example.ipynb [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+    "plt.hist(rare_data_np, weights=rare_weights_np, bins=40)\n",
+    "\n",
+    "# create the model to fit\n",
+    "\n",
+    "# set the normalization range of the exponential to the whole range\n",
+    "comb_bkg_rare.set_norm_range(obs)\n",
+    "\n",
+    "# parameters for the model\n",
+    "mu = zfit.Parameter('mu', 5270, 5200, 5350)\n",
+    "sigma = zfit.Parameter('sigma', 22, 0, 100)\n",
```
Comment (jonas-eschle) [inline 06-ecosystem-example.ipynb]:
```suggestion
    "mu = zfit.Parameter('mu', 5270, 5200, 5350, step_size=3)\n",
    "sigma = zfit.Parameter('sigma', 22, 0, 100, step_size=0.5)\n",
```

File: 06-ecosystem-example.ipynb [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+    "# ------------------------------------------------\n",
+    "# RESONANT MODEL\n",
+    "# ------------------------------------------------\n",
+    "\n",
+    "# create the model to fit\n",
+    "\n",
+    "# we can share parameters directly or create composed parameters. Here we have a\n",
+    "# parameter that scales the sigma from the rare fit\n",
+    "\n",
+    "sigma_scaling = zfit.Parameter('sigma_scaling', 0.9, 0.1, 10)\n",
```
Comment (jonas-eschle) [inline 06-ecosystem-example.ipynb]:
```suggestion
    "sigma_scaling = zfit.Parameter('sigma_scaling', 0.9, 0.1, 10, step_size=0.1)\n",
```

**Review by jonas-eschle — CHANGES_REQUESTED** [review]
This should do the trick

**Comment by eduardo-rodrigues** [comment]
Thanks a lot @mayou36.

BTW, have you thought of adding hints when the failures observed happen, which are likely to get sorted with this step_size parameter? The stacktrace is not obvious.

**Comment by jonas-eschle** [comment]
Yes, and actually there is a message saying this. But it is buried under TensorFlow warnings...

**Comment by eduardo-rodrigues** [comment]
Oops, I read too quick. Thanks.

**Pull request merged**